.orchestra/support/distributable-test.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import io
+import os
+import socket
+import sys
+import traceback
+from graphlib import TopologicalSorter
+from subprocess import Popen
+from time import sleep
+from urllib.error import URLError
+from urllib.request import urlopen
+
+from gql import Client, gql
+from gql.transport.requests import RequestsHTTPTransport
+from psutil import Process
+
+FILTER_ENV = [
+    "STARLETTE_DEBUG",
+    "REVNG_NOTIFY_FIFOS",
+    "REVNG_ORIGINS",
+    "REVNG_DATA_DIR",
+    "REVNG_PROJECT_ID",
+]
+
+
+def print_fd(fd: int):
+    os.lseek(fd, 0, io.SEEK_SET)
+    out_read = os.fdopen(fd, "r")
+    print("\n\n Daemon Stacktrace: \n")
+    print(out_read.read())
+
+
+def check_server_up(address: str):
+    for _ in range(10):
+        try:
+            req = urlopen(f"http://{address}/status", timeout=1.0)
+            if req.code == 200:
+                return
+            sleep(1.0)
+        except (URLError, TimeoutError, ConnectionResetError, socket.timeout):
+            sleep(1.0)
+    raise ValueError()
+
+
+def get_listen_port(pid: int) -> int:
+    psutil_process = Process(pid)
+    while True:
+        connections = psutil_process.connections()
+        for connection in connections:
+            if connection.raddr == () and connection.status == "LISTEN":
+                return connection.laddr.port
+        sleep(0.5)
+
+
+def main():
```

Comment:
This belongs to the revng project, or it will be easier to make it go out of sync with revng (e.g., changes int the GraphQL API). Maybe we can have a `revng self-test` command?

Reply:
I've added in the sister PR for revng the command `revng daemon-self-test`